plugins/output/http.md

```diff
@@ -282,6 +292,79 @@ log line4\n
 ...
 ```
 
+## Troubleshooting
+
+### 400 Bad request between out\_http and in\_http
+
+When getting the following error:
+
+```
+#0 got unrecoverable error in primary and no secondary error_class=Fluent::UnrecoverableError error="400 Bad Request 400 Bad Request\n'json' or 'msgpack' parameter is required\n"
+#0 bad chunk is moved to /tmp/fluent/backup/worker0/object_3ff8a73edae8/5a71a08ca19b1b343c8dce1b74c9a963.log
+```
+
+Users should be specify `json` format with `json_array` as true for out\_http configuration:
+
+```
+<match **>
+  @type http
+  endpoint http://some.your.http.endpoint:9811/your-awesome-path
+  <format>
+    @type json
+  </format>
+  json_array true
+  <buffer>
+    flush_interval 2s
+  </buffer>
+</match>
+```
+
+And receiver in\_http configuration should be:
+
+```
+<source>
+  @type http
+  port 9811
+  bind 0.0.0.0
+  <parse>
+    @type json
+  </parse>
+</source>
+```
+
+Or specify msgpack format:
+
+```
+<match **>
+  @type http
+  endpoint http://some.your.http.endpoint:9882/your-awesome-path
+  <format>
+    @type msgpack
+  </format>
+  <buffer>
+    flush_interval 2s
+  </buffer>
+</match>
+```
+
+And receiver in\_http configuration should be:
+
+```
+<source>
+  @type http
+  port 9882
+  bind 0.0.0.0
+  <parse>
+    @type msgpack
+  </parse>
+  <format>
+    @type json
+  </format>
+</source>
+```
+
+But, we recomment to use in/out [`forward`](forward.md) plugin to communicate with two Fluentds due to at-most-once and at-least-once semantics for rigidty.
```

Comment:
type: recommend

Reply:
Oops. 😖

Reply:
Fixed in https://github.com/fluent/fluentd-docs-gitbook/pull/184/commits/e2d21f1505a3246fb3d1602adba1b121372d500c.